From 478ceb73a1ca3ccdef48690310903bea06cc17b6 Mon Sep 17 00:00:00 2001
From: Harish Subramony <harish.subramony@intel.com>
Date: Tue, 30 Jun 2015 11:58:37 -0700
Subject: [PATCH] HSU: Crash in HSU driver on boot

Fix issue with patch 8347af22d512cf303c5920f6053600a2311ef1cf
if RTS/CTS gpios are not defined in the ACPI table.

Change-Id: I2a9925208d1d0d99e3532b72ca8e4ba7376eaf30
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-40689
Signed-off-by: Harish Subramony <harish.subramony@intel.com>
Signed-off-by: Huiquan Zhong <huiquan.zhong@intel.com>
---
 drivers/tty/serial/hsu/hsu_plat.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/tty/serial/hsu/hsu_plat.c b/drivers/tty/serial/hsu/hsu_plat.c
index 8069f2d8c70e..f68cccae9a63 100644
--- a/drivers/tty/serial/hsu/hsu_plat.c
+++ b/drivers/tty/serial/hsu/hsu_plat.c
@@ -54,6 +54,8 @@ static int cht_hw_set_rts(struct uart_hsu_port *up, int value)
 		if (!pin_cfg->rts_gpio) {
 			pin_cfg->rts_gpio = gpiod_get_index(up->dev, "hsu_rts",
 					hsu_rts_idx);
+			if (IS_ERR(pin_cfg->rts_gpio))
+				pin_cfg->rts_gpio = NULL;
 		}
 
 		if (pin_cfg->rts_gpio) {
@@ -84,6 +86,8 @@ static int cht_hsu_hw_suspend(struct uart_hsu_port *up)
 		if (!pin_cfg->rx_gpio) {
 			pin_cfg->rx_gpio = gpiod_get_index(up->dev, "hsu_rxd",
 					hsu_rxd_idx);
+			if (IS_ERR(pin_cfg->rx_gpio))
+				pin_cfg->rx_gpio = NULL;
 		}
 		pin_cfg->wake_gpio = pin_cfg->rx_gpio;
 		break;
@@ -91,6 +95,8 @@ static int cht_hsu_hw_suspend(struct uart_hsu_port *up)
 		if (!pin_cfg->cts_gpio) {
 			pin_cfg->cts_gpio =gpiod_get_index(up->dev, "hsu_cts",
 					hsu_cts_idx);
+			if (IS_ERR(pin_cfg->cts_gpio))
+				pin_cfg->cts_gpio = NULL;
 		}
 		pin_cfg->wake_gpio = pin_cfg->cts_gpio;
 		break;
-- 
1.9.1

